gateway/handle.go

```diff
@@ -759,71 +768,56 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(
 			continue
 		}
 
-		userID := buildUserID("", anonIDFromReq, userIDFromReq)
-		receivedAt, _ := toSet["receivedAt"].(string)
+		receivedAt := gjson.GetBytes(msg.Payload, "received_at").String()
 		if receivedAt == "" {
 			receivedAt = time.Now().Format(misc.RFC3339Milli)
 		}
-		out = append(out, jobObject{
-			userID: userID, events: []map[string]interface{}{toSet}, receivedAt: receivedAt,
-		})
-	}
-	if len(out) == 0 { // events suppressed - but return success
-		return nil, nil
-	}
-	var params struct {
-		SourceID        string `json:"source_id"`
-		SourceJobRunID  string `json:"source_job_run_id"`
-		SourceTaskRunID string `json:"source_task_run_id"`
-	}
-	params.SourceID = sourceID
-	params.SourceJobRunID = sourcesJobRunID
-	params.SourceTaskRunID = sourcesTaskRunID
-	marshalledParams, err := json.Marshal(params)
-	if err != nil {
-		gw.logger.Errorn(
-			"[Gateway] Failed to marshal parameters map. Parameters: %+v",
-			logger.NewField("params", params),
-			obskit.Error(err),
-		)
-		marshalledParams = []byte(
-			`{"error": "rudder-server gateway failed to marshal params"}`,
-		)
-	}
 
-	jobs := make([]*jobsdb.JobT, 0, len(out))
-	type singularEventBatch struct {
-		Batch      []map[string]interface{} `json:"batch"`
-		RequestIP  string                   `json:"requestIP"` // update processor accordingly
-		WriteKey   string                   `json:"writeKey"`
-		ReceivedAt string                   `json:"receivedAt"`
-	}
-	for _, userEvent := range out {
-		var (
-			payload    json.RawMessage
-			eventCount int
-		)
+		jobsDBParams := params{
+			MessageID:       msg.Properties.MessageID,
+			SourceID:        msg.Properties.SourceID,
+			SourceJobRunID:  msg.Properties.SourceJobRunID,
+			SourceTaskRunID: msg.Properties.SourceTaskRunID,
+			UserID:          msg.Properties.UserID,
+			TraceParent:     msg.Properties.TraceID,
+		}
+
+		marshalledParams, err := json.Marshal(jobsDBParams)
+		if err != nil {
+			gw.logger.Errorn(
+				"[Gateway] Failed to marshal parameters map. Parameters: %+v",
+				logger.NewField("params", jobsDBParams),
+				obskit.Error(err),
+			)
+			marshalledParams = []byte(
+				`{"error": "rudder-server gateway failed to marshal params"}`,
+			)
+		}
+
 		eventBatch := singularEventBatch{
-			Batch:      userEvent.events,
-			WriteKey:   arctx.WriteKey,
-			ReceivedAt: userEvent.receivedAt,
+			Batch:      []json.RawMessage{msg.Payload},
+			ReceivedAt: receivedAt,
 		}
-		payload, err = json.Marshal(eventBatch)
+
+		payload, err := json.Marshal(eventBatch)
 		if err != nil {
-			panic(err)
+			return nil, err
```

Comment:
Can we add some context here?
```suggestion
			return nil,  fmt.Errorf("marshalling event batch: %w", err)
```